### PR TITLE
Add documentation examples for character/complete

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #914.

### CODE REVIEW

File: src/character/complete.rs [hunk 1]
```diff
@@ -9,9 +9,21 @@ use traits::{AsChar, AtEof, FindToken, InputIter, InputLength, InputTakeAtPositi
...
-/// *complete version*: will return an error if there's not enough input data
+/// *complete version*: Will return an error if there's not enough input data.
+///
+/// # Example
+///
+/// ```
+/// # use nom::{Err, error::ErrorKind};
+/// # use nom::character::complete::char;
+/// # fn main() {
+/// assert_eq!(char::<_, (&str, ErrorKind)>('a')("abc"), Ok(("bc", 'a')));
```
Comment (Geal) [inline src/character/complete.rs]: It's not a good idea to write `char::<_, (&str, ErrorKind)>` everywhere, it makes the example hard to read. If type inference on the error type is an issue, it should be better to write:
```rust
fn parser(input: &str) -> IResult<&str, &str> {
  char('a')
}
```

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by GuillaumeGomez** [comment]
So after some investigations, I'm wondering if this isn't coming from rustdoc... For records, the error output is:

```
---- src/character/complete.rs - character::complete::alphanumeric1 (line 770) stdout ----
thread 'src/character/complete.rs - character::complete::alphanumeric1 (line 770)' panicked at 'test executable failed:

thread 'main' panicked at 'not yet implemented: error[E0658]: use of unstable library feature 'core_char_ext': the stable interface is `impl char` in later crate (see issue #32110)', src/traits.rs:258:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::_print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::continue_panic_fmt
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: <char as nom::traits::AsChar>::is_alpha
   9: <char as nom::traits::AsChar>::is_alphanum
  10: nom::character::complete::alphanumeric1::{{closure}}
  11: <&str as nom::traits::InputTakeAtPosition>::split_at_position1_complete::{{closure}}
  12: core::iter::traits::iterator::Iterator::find::{{closure}}
  13: core::iter::traits::iterator::Iterator::try_for_each::{{closure}}
  14: core::iter::traits::iterator::Iterator::try_fold
  15: core::iter::traits::iterator::Iterator::try_for_each
  16: core::iter::traits::iterator::Iterator::find
  17: <&str as nom::traits::InputTakeAtPosition>::split_at_position1_complete
  18: nom::character::complete::alphanumeric1
  19: rust_out::main::parser
  20: rust_out::main
  21: std::rt::lang_start::{{closure}}
  22: std::panicking::try::do_call
  23: __rust_maybe_catch_panic
  24: std::rt::lang_start_internal
  25: std::rt::lang_start
  26: main

', src/librustdoc/test.rs:332:17
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::_print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::continue_panic_fmt
   6: std::panicking::begin_panic_fmt
   7: <rustdoc::test::Collector as rustdoc::test::Tester>::add_test::{{closure}}
   8: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
   9: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
  10: __rust_maybe_catch_panic
  11: test::run_test::run_test_inner::{{closure}}
```

Funny thing is that it works perfectly fine when doing the same thing in a binary alongside. I recall you, @QuietMisdreavus working on rustdoc's test handling recently. Do you think this is coming from the same part?

**Comment by QuietMisdreavus** [comment]
Looking at it a bit deeper, it looks like that builder is building `nom` without the `alloc` feature, which causes the `is_alpha` function called by those doctests to trip this assertion:

https://github.com/Geal/nom/blob/c003a2102bb2a4d18d80f529d86bba6e214eccc6/src/traits.rs#L258-L260

Should those functions be gated on the `alloc` feature?

**Comment by GuillaumeGomez** [comment]
As I understood, @Geal said that I wasn't doing something correctly on my side, but maybe you're right. I'll go for that for the moment and then we'll see. Thanks!

**Comment by Geal** [comment]
merged in 2ad650940545

**Pull request closed without merging**